mirrord-agent/src/main.rs

```diff
@@ -94,6 +97,34 @@ struct PeerMessage {
     peer_id: PeerID,
 }
 
+async fn handle_peer_message(
+    message: PeerMessage,
+    tx: mpsc::Sender<PeerMessage>,
+    stream: &mut actix_codec::Framed<TcpStream, DaemonCodec>,
+) -> Result<()> {
+    match message.msg {
+        // NOTE(alex): Handling file requests here for simplicity.
+        ClientMessage::OpenFileRequest(path) => {
+            debug!(
+                "handle_peer_message -> peer id {:?} asked to open file {path:?}",
+                message.peer_id
+            );
+
+            let file = std::fs::File::open(path)?;
+            let file_fd = file.as_raw_fd();
+
+            debug!("handle_peer_message -> file is open with fd {file_fd:?}");
+
+            let open_file_message =
+                DaemonMessage::OpenFileResponse(FileOpenResponse { fd: file_fd });
+            stream.send(open_file_message).await?;
```

Comment:
You should call open, then if there's an error return the error, if it's success you just save the fd (the layer doesn't need the actual fd that the agent has)